### PR TITLE
[REF] note,project: use the project.task model for notes/todos

### PR DESCRIPTION
**Pull request by hugocarlier**
Module note is rewritten to be based on the model project.task. This model is then extended in the module project. This will allow to have a similar behavior between tasks and notes/todos and to be able to convert easily a note/todo into a task in the project app.

Model harmonization
===================

As the goal of this conversion of the note app is to be able to convert easily a note to a task, notes and tasks are now based on the same model (project.task). This model is created in the module note and overridden in the module project. The same logic would be followed for project.tags and project.task.type models that are used by project.task.

It is therefore decided that the module note is requested to use the project app and would be auto-installed when this later in installed.

The following fields are created in the different models: 
(field with a * serve a similar purpose as an old field but with a different name and/or type)
(field with a ** are new in this module)

|                   | To-do                                                                                                                                                                        | Project                                                                                                                                                                                                   |
|-------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| project.tags      | - name<br> - color<br><br> <br>                                                                                                                                                                | - name<br> - color<br> - project_ids<br> - task_ids                                                                                                                                                                       |
| project.task.type | - name<br> - sequence<br> - user_id<br> - fold<br> - todo_stage (Bool)**<br> <br> <br>                                                                                                                        | - name<br> - sequence<br> - user_id<br> - fold<br> - todo_stage (Bool)**<br> - project_ids<br> - ...                                                                                                                                     |
| project.task      | - name (Char*)<br> - company_id<br> - user_id<br> - description*<br> - sequence<br> - personal_stage_type_ids*<br> - personal_stage_type_id*<br> - active*<br> - color<br> - tag_ids<br> - message_partner_id<br> - is_todo (Bool)**<br><br><br><br><br> | - name<br> -company_id<br> - user_id**<br> - description<br> - sequence<br> - personal_stage_type_ids<br> - personal_stage_type_id<br> - active<br> - color<br> - tag_ids<br> - message_partner_id**<br> - is_todo (Bool)**<br> - user_ids<br> - project_id<br> - stage_id<br> - ... |

Details:
- todo: 
    - memo is replaced by description (note content) 
    - stage_ids is replaced with personal_stage_type_ids 
    - stage_id is replaced with personnal_stage_type_id 
    - open is replaced by active 
    - field date_done is currently not used in the Note app and is therefore removed
- todo and project: 
    - todo_stage is added in project.task.type to distinguish stages in project and in todo 
    - is_todo is added in project.task to dinstinguish tasks in project and in todo 
    - message_partner_ids will have the attribute compute_sudo=True that was set to False previously in project 
    - field personal_stage_id and model project.task.stage.personal are not necessary for the management of personnal stage and would be deleted (to investigate).

Personal stages
===============

- The logic to manage personal stages in project and note modules is harmonized. The main changes and the reason that led to it are the following:

    - project.trask kanban view: records are grouped by
    personal_stage_type_ids (M2M) (instead of stage_id (M2O) previously
    in note) (as it is currently the case in the project module). This
    allows:
        - better code clarity: we do not need to override completely the
        read_group method anymore.
        - better performances: Loading time divided by ~3 for a large
        number of records.
    - The management of missing personal stages for a user is now
    managed at this user level. The allows to distribute the computation
    time between every users instead of grouping it at records
    writing/app installation and therefore leads to better
    performances. This is done by:
        - Default personal stages are created when the user load the
        kanban view in the Note app/the kanaban view of "My
        Tasks" (instead of at app installation/user creation currently
        for the module note).
        - Missing personal stage for a task is assigned for a given user
        when this user load the kanban view where it appears. In project
        it was done at the writing of new assignees for a task.

Tags
====

A simple flag todo_tag (Boolean) is added to the model project.tags. This allow to:
- dinstinguish project tags from note tags (and manage the difference in access rules between the
two types of tags)
- Avoid adding an additional M2M field to the model project.task.

The same access rules as previously used are replicated, i.e:
- In Project, only administrator can create and edit tags.
- In Todo, all users can create and edit any tags.
- No user should have access to project tag from Todo and vice-versa.

To be implemented : When a note/todo is converted to a task, a match of
the current task of the note with existing tags in project can be
performed to replace its tags. However, no project tags can be created
based on the current tags of the note, because it will give a way for
any user to create tags in project (which should be for project manager
only).

Security
========

- Access rights: project.task model need to be accessible in all crud
modes to all internal users as it is used in the note module. We avoid
to extend existing rights in projects by setting the exisiting record
rules for the internal user group only for read mode.

- Record rules: The flag is_todo is used to grant access to records in
the note module. The philosophy is to change as few rules as possible in
the project module as they have a big impact on existing databases. They
mostly remains the same with no use of the flag is_todo as they do not
grant additional access in note module.

- In summary:
    - Default access rights to project.task is managed in note module
    - Global record rules are declared in note module
    - Records rules in project apply to both todo and tasks in project
    - Extension of the access for todo should be done using the flag
    is_todo

### CODE REVIEW

**Comment by hugho-ad** [comment]
Notes/todos 

for me they are a tasks with an específic tag instead
